lib/sax/parser_feedback_simulator.js

```diff
@@ -76,13 +89,31 @@ ParserFeedbackSimulator.prototype._ensureTokenizerMode = function (tn) {
 ParserFeedbackSimulator.prototype._handleStartTagToken = function (token) {
     var tn = token.tagName;
 
-    if (tn === $.SVG)
+    if (tn === $.SVG) {
         this._enterNamespace(NS.SVG);
+        foreignContent.adjustTokenSVGAttrs(token);
+        foreignContent.adjustTokenXMLAttrs(token);
+    }
 
     else if (tn === $.MATH)
         this._enterNamespace(NS.MATHML);
 
+    else if (tn === $.IMAGE)
+        token.tagName = $.IMG;
```

Comment:
Btw, theoretically, if we get close enough, we could even use this simulator in between the real parser and tokenizer in order to avoid logic duplication and share more code between non- and streaming parsers.